packages/composites/token-allocation/src/dataProvider.ts

```diff
@@ -25,97 +25,95 @@ import { AdapterRequest } from '@chainlink/types'
  * ```
  */
 
-export const getPriceProvider =
-  (source: string, jobRunID: string, apiConfig: RequestConfig) =>
-  async (symbols: string[], quote: string, withMarketCap = false): Promise<ResponsePayload> => {
-    if (source === 'coinpaprika') {
-      return await sendBatchedRequests(source, jobRunID, apiConfig, symbols, quote, withMarketCap)
-    }
-    return await sendIndividualRequests(source, jobRunID, apiConfig, symbols, quote, withMarketCap)
+export const getPriceProvider = (
+  source: string,
+  jobRunID: string,
+  apiConfig: RequestConfig,
+): GetPrices => {
+  if (source === 'coinpaprika') {
+    return sendBatchedRequests(source, jobRunID, apiConfig)
```

Comment:
Why is `coinpaprika` special?

Reply:
This was done ~2 weeks ago for performance reasons. The way that Coinpaprika handles batching involves getting back a 10MB payload of data. If Token Allocations has 50 pairs to get prices of and sends them one by one it causes a huge spike.

Reply:
Fair enough. In a perfect world I'd like to avoid this hard-coded conditional logic, but I understand why in this special circumstance.